go/ql/lib/semmle/go/frameworks/AwsLambda.qll

```diff
@@ -0,0 +1,59 @@
+/**
+ * Provides classes for working with untrusted flow sources, sinks and taint propagators
+ * from the `github.com/aws/aws-lambda-go/lambda` package.
+ */
+
+import go
+
+/** A source of input data in an AWS Lambda. */
+private class LambdaInput extends UntrustedFlowSource::Range {
+  LambdaInput() {
+    exists(Parameter p | p = this.asParameter() |
+      p = any(HandlerFunction hf).getAParameter() and
+      not p.getType().hasQualifiedName("context", "Context") and
+      not p instanceof ReceiverVariable
+    )
+  }
+}
+
+private class HandlerFunction extends FuncDef {
+  HandlerFunction() {
+    exists(StartOrNewHandlerFunction f, DataFlow::Node handlerArg |
+      f.getACall().getArgument(f.getHandlerArgPos()) = handlerArg
+    |
+      handlerArg = this.(FuncDecl).getFunction().getARead() or
+      handlerArg = DataFlow::exprNode(this.(FuncLit))
+    )
+    or
+    this = any(Method m | m.implements(awsLambdaPkg(), "Handler", "Invoke")).getFuncDecl()
+    or
+    exists(ConversionExpr ce |
+      ce.getTypeExpr().getType() instanceof HandlerImpl and
+      this = ce.getOperand().(FunctionName).getTarget().getFuncDecl()
```

Comment:
```suggestion
    exists(TypeCastNode typeCast |
      typeCast.getResultType() instanceof HandlerImpl and
      this.(FuncDecl).getFunction().getARead() = typeCast.getOperand()
```

Sorry, should have picked this one up at the same time as the above. Also are there any tests for this path at the moment?

Reply:
👍 

> Also are there any tests for this path at the moment?

Yes, the ones using `MyHandlerFunc`, e.g.

```go
lambda.StartHandler(MyHandlerFunc(Handler9))
```